backend/apps/question/src/question.service.ts

```diff
@@ -50,30 +54,53 @@ export class QuestionService {
   async addQuestion(
     questionWithCategoriesAndDifficulty: QuestionWithCategoryAndDifficulty,
   ): Promise<QuestionWithCategoryAndDifficulty> {
-    // Check if difficulty and categories exist
-    const difficultyObject = await this.getDifficultyIfExists(
-      questionWithCategoriesAndDifficulty.difficulty,
-    );
-    const categoryObjects = await Promise.all(
-      questionWithCategoriesAndDifficulty.categories.map((category) =>
-        this.getCategoryIfExists(category),
-      ),
-    );
+    try {
```

Comment:
Can we abstract this try-catch logic into a decorator? e.g. https://javascript.plainenglish.io/a-try-catch-decorator-to-stylize-your-code-bdd0202765c8

Reply:
Done!